languages/dart/keywords/abstract.md

```diff
@@ -0,0 +1,102 @@
+# Abstract
+
+The keyword `abstract` is used to define an *abstract class* - a [class][reference-class] that can not be instantiated 
+directly, which means you can not create an [object][reference-object] from it. Conversely, a class that can be 
+instantiated directly is called a *concrete class*.
+
+Abstract types are [interfaces][reference-interface] that hide the implementation details and show only functionality to
```

Comment:
This line is not completely true. Instead, I would something along the lines of:

`Dart does not have [interfaces][reference-interface], a type that hides implementation details and show only functionality to the user. They let you focus on **what** the object does instead of **how** it does it.

Instead Dart has abstract types which can be used as interfaces, but do allow any and all methods to be implemented. So it's up to the user how much implementation, if any, is in the abstract class.`

Reply:
While it's true dart no longer has [inteface](https://news.dartlang.org/2012/06/proposal-to-eliminate-interface.html) keyword, It still has the concept of interface, I believe what @Corentin-Leffy wrote, was indeed correct. The official language guide does say the same thing, [abstract-classes](https://dart.dev/guides/language/language-tour#abstract-classes), it also explains the [implicit-interface](https://dart.dev/guides/language/language-tour#implicit-interfaces) nature of it's type system, whenever you define a type (including mixins) it's an interface in dart, and this is why I wanted to include some thoughts on `implements` keyword here as well. When using the keyword `implements` instead of `extends` we *explicitly* use a certain type (including mixins) as an interface.

Saying that dart doesn't have interfaces while the official guide says it does (implicitly) and refers to the term "interface" multiple times will only cause confusion.

Thoughts?

<img src="https://i.imgflip.com/3nbjn8.jpg" title="made at imgflip.com"/>

Reply:
@devkabiir I saw that in the Dart docs, but I think the best place to address that is in the class keyword file. We can't describe everything here.

If you want, we can rephrase this file to say that all Dart classes are implicit interfaces and then link to the class keyword file.

Reply:
Sounds good to me.